test/testdata/infer/generics/variance_methods.rb

```diff
@@ -0,0 +1,45 @@
+# typed: true
+
+module A
+  extend T::Sig
+  extend T::Helpers
+  extend T::Generic
+
+  abstract!
+
+  In = type_member(:in)
+  Out = type_member(:out)
+
+  # should pass: the parameters are used correctly
+  sig {abstract.params(key: In).returns(Out)}
+  def lookup(key); end
+
+  # should pass: the parameters are used correctly, and the block argument swaps
+  # the polarity of the type members
+  sig {abstract.params(key: In, blk: T.proc.params(arg: Out).returns(In)).returns(Out)}
```

Comment:
Do you want to try going through a variant generic class other that Proc?

Reply:
I've added the `variance_user.rb` test to test variance with a user-defined type.